asyncify.mjs

```diff
@@ -48,10 +36,23 @@ function proxyGet(obj, transform) {
   });
 }
 
-class Asyncify {
-  constructor() {
+export class Asyncify {
+  constructor(options) {
+    options = options || {}
     this.value = undefined;
     this.exports = null;
+
+    // Put `__asyncify_data` somewhere at the start.
+    // The default address is pretty hand-wavy
+    // See https://github.com/WebAssembly/binaryen/blob/6371cf63687c3f638b599e086ca668c04a26cbbb/src/passes/Asyncify.cpp#L106-L113
+    // for structure details.
+    this.dataAddr = options.dataBegin || 16;
+    // Place actual data right after the descriptor (which is 2 * sizeof(i32) = 8 bytes).
+    this.dataStart = this.dataAddr + 8;
+    // End data at 1024 bytes. This is where the unused area by Clang ends and real stack / data begins.
+    // Because this might differ between languages and parameters passed to wasm-ld, ideally we would
+    // use `__stack_pointer` here, but, sadly, it's not exposed via exports yet.
+    this.dataEnd = options.dataEnd || 1024;
```

Comment:
I think we'd want to do something different here as well - either throw an error, or accept `dataSize` and calculate end address ourselves.

E.g. if user provided custom `dataStart` but not `dataEnd`, then it doesn't make sense to fallback to `1024`.

Reply:
Thinking about it a bit more, throwing would be most appropriate - that is, if user provided `dataStart`, then ensure that `dataEnd` is also present and is `>= dataStart`.

Reply:
Ok thanks for all the feedback on this and the fast responses, tried to capture - let me know what you think.